### ISSUE TITLE
Rule Proposal: Consistent spacing for multiline list items

### ISSUE THREAD

**Issue by guillp** [rvben/rumdl#682]
### Problem or motivation

A rule could check that all multiline items in a list are consistently spaced, e.g.

```markdown
# test for inconsistent multiline list items spacing

- this is a long line
that continues on a second line
- this is another long line
  that continues on the next line
- yet again a long line
and still inconsistently spaced
  and even worse
```

...currently raises no warning, while it should be consistently formatted like this:

```markdown
# test for inconsistent multiline list items spacing

- this is a long line
  that continues on a second line
- this is another long line
  that continues on the next line
- yet again a long line
  and still inconsistently spaced
  and even worse
```

or like this:
```
# test for inconsistent multiline list item spacing

- this is a long line
that continues on a second line
- this is another long line
that continues on the next line
- yet again a long line
and still inconsistently spaced
and even worse
```

### Proposed solution

- [MD077](https://rumdl.dev/md077) is related but not exactly the same: it checks each list item to make sure that continuation lines have either 0 leading spaces or just enough to pad them to the beginning of the first line.
- [MD005](https://rumdl.dev/md005/) is also related but it only checks the indentation for the first line of a list item, not the next ones.
So I suggest a rule (MD082?) to check that multiple items in the same list are spaced consistently. I'll can try to propose a PR if approved.

### Feature category

New lint rule

### Example markdown

```markdown
- this is a long line
that continues on a second line
- this is another long line
  that continues on the next line
- yet again a long line
and still inconsistently spaced
  and even worse
 and
```

### Contribution

- [x] I would be willing to submit a PR to implement this feature

### Additional context

I did not find an equivalent rule in `markdownlint`.
`mdformat` enforces this rule by indenting every item consistently to the same level as the first line content.

To make it autofixable, a decision must be made to either:
- deindent every continuation line to 0 space
- or indent each enough to match the first line, like `mdformat`

which is a mater of personal preferences, so it should be configurable. What should be the default? 
I considered either mdformat style, or to align to the majority-style in each list block.

**Comment by guillp** [comment]
Something I forgot: how to handle sublist items:

```markdown
* List Item 1
  * SubList Item 1-1
  with multiple lines
  * SubList Item 1-2
    with multiple lines
  * SubList Item 1-3
with multiple lines
```

This could be formatted consistently in 3 different ways:
```markdown
* List Item 1
  * SubList Item 1-1
with multiple lines
  * SubList Item 1-2
with multiple lines
  * SubList Item 1-3
with multiple lines
```

Aligned to first line content:
```markdown
* List Item 1
  * SubList Item 1-1
    with multiple lines
  * SubList Item 1-2
    with multiple lines
  * SubList Item 1-3
    with multiple lines
```

Or aligned to the first line list sign:
```markdown
* List Item 1
  * SubList Item 1-1
  with multiple lines
  * SubList Item 1-2
  with multiple lines
  * SubList Item 1-3
  with multiple lines
```

**Comment by rvben** [comment]
Hi @guillp, thanks for the suggestion! Will look into what makes most sense and if/how it can fit in.

**Comment by guillp** [comment]
@rvben: thanks for the very quick implementation! 
With the new config option `MD077.style = "aligned"`, `rumdl` found and fixed everything I expected in my doc base. I would consider this the "best" style indeed, to the point it should be default.
This even helped me spot instances of missing blank lines around lists, like:

```markdown
- list item 1
- list item 2
Line of text to introduce another list but missing a blank line to separate from the list above:
- another list item 
```
rendered as:

---
- list item 1
- list item 2
Line of text to introduce another list but missing a blank line to separate from the list above:
- another list item 
---
which has a different semantic (in standard markdown at least) than:

```markdown
- list item 1
- list item 2

Line of text to introduce another list but missing a blank line to separate from the list above:
- another list item 
```
rendered as:

---
- list item 1
- list item 2

Line of text to introduce another list but missing a blank line to separate from the list above:
- another list item 

---
(and I notice now that GitHub doesn't render this the same as standard markdown)

So many thanks for this and the whole `rumdl`!

**Comment by rvben** [comment]
Thanks for the request, and the feedback! It really helps in improving rumdl for everyone 🙂💪

**Comment by rvben** [comment]
Covered by MD077's opt-in `aligned` style:

```toml
[MD077]
style = "aligned"
```

Requires every continuation line to align to its item's content column; `--fix` snaps them into place. Confirmed it handles your nested-sublist case (aligns each continuation to its sub-item's content column).

Closing as addressed by the existing rule. Making `aligned` the default is a separate call; open a new issue if you want that.
